src/commands/reword.rs

```diff
@@ -0,0 +1,354 @@
+                writeln!(
+                    effects.get_output_stream(),
+                    "Reworded {} commits with same message. If this was unintentional, run: git undo",
+                    num_commits,
+                )?;
+            }
+
+            0
+        }
+        ExecuteRebasePlanResult::DeclinedToMerge { merge_conflict: _ } => {
+            writeln!(
+                effects.get_error_stream(),
+                "BUG: Merge conflict detected, but rewording shouldn't cause any conflicts."
+            )?;
+
+            1
+        }
+        ExecuteRebasePlanResult::Failed { exit_code } => exit_code,
+    };
+
+    Ok(exit_code)
+}
+
+/// Turn a list of ref-ish strings into a list of Commits.
+fn resolve_commits_from_hashes<'repo>(
+    repo: &'repo Repo,
+    dag: &mut Dag,
+    effects: &Effects,
+    hashes: Vec<String>,
+) -> eyre::Result<Option<Vec<Commit<'repo>>>> {
+    let hashes = if hashes.is_empty() {
+        vec!["HEAD".to_string()]
+    } else {
+        hashes
+    };
+
+    let commits = resolve_commits(effects, repo, dag, hashes)?;
+    let commits = match commits {
+        ResolveCommitsResult::Ok { commits } => commits,
+        ResolveCommitsResult::CommitNotFound { commit: hash } => {
+            writeln!(effects.get_output_stream(), "Commit not found: {}", hash)?;
+            return Ok(None);
+        }
+    };
+    Ok(Some(commits))
+}
+
+/// The result of building the reword message.
+#[must_use]
+pub enum BuildRewordMessageResult {
+    /// The reworded message was built successfully.
+    Succeeded {
+        /// The reworded messages for each commit.
+        messages: HashMap<NonZeroOid, String>,
+    },
+
+    /// The reworded message matches the original message.
+    IdenticalMessage,
+
+    /// The reworded message was empty.
+    EmptyMessage,
+}
+
+/// Builds the message(s) that will be used for rewording. These are mapped from each commit's
+/// NonZeroOid to the relevant message.
+#[instrument]
+fn build_messages(
+    messages: &[String],
+    commits: &[Commit],
+    comment_char: Option<u8>,
+) -> eyre::Result<BuildRewordMessageResult> {
+    let message = messages.join("\n\n").trim().to_string();
+
+    let (message, load_editor) = if message.is_empty() {
+        let message_for_editor = match commits.to_vec().as_slice() {
+            [commit] => match commit.get_message_raw()?.into_string() {
+                Ok(msg) => msg,
+                _ => eyre::bail!(
+                    "###
+                    Error decoding original message for commit: {:?}.
+                    Stopping; nothing reworded.
+                    ###",
+                    commit.get_oid()
+                ),
+            },
+            _ => {
+                // TODO(bulk edit) build a bulk edit message for multiple commits
+                format!(
+                    "Enter a commit message to apply to {} commits",
+                    commits.len()
+                )
```

Comment:
It occurs to me that we might want to have a more descriptive default message. Perhaps we should be appending something like the default commit message when prepping the string for the Editor:

```
# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
```

This would be easy to adapt from `commit` to `reword`.

Reply:
I added this "help" message to Editor mode, and I'm glad I did: I had been *completely* misunderstanding how Rust handles newlines and whitespace in strings. I was using some sort of `"### ... ###"` syntax based on the snapshots in the tests b/c I thought that was some special Rust syntax to "dedent" those strings (ie remove leading whitespace from each line), but hadn't realized that it's just `insta` that's doing that. 

That was all sorts of wrong and I've rewritten my multi-line strings using `\n\` syntax, which actually achieves what I was going for initially.